src/interrupts/gtd.rs

```diff
@@ -58,5 +61,8 @@ pub fn init_gdt() {
     unsafe {
         CS::set_reg(GDT.1.code_selector);
         load_tss(GDT.1.tss_selector);
+        SS::set_reg(GDT.1.data_selector);
```

Comment:
![image](https://user-images.githubusercontent.com/1353942/163725449-b2b45d39-dd7e-409f-9101-24a5f479c9dd.png)
![image](https://user-images.githubusercontent.com/1353942/163725525-954abeaa-1074-4d90-80d4-70c832471737.png)

Those segment descriptors area only added because the CPU checks them when loading the GDT after an interrupt.
It probably just needs to have a PRESENT flag set for them and that's all.

Reply:
alright